File: Models/Item.cs
#nullable disable

using System.Text.Json.Serialization;

namespace WebPage.Models
{
    public class Item
    {
        public ItemTypes type { get; set; }
        [JsonPropertyName("id")]
        public int Id { get; set; }
        [JsonPropertyName("name")]
        public string Name { get; set; }
        [JsonPropertyName("compositions")]
        public List<CompositionElement> Composition { get; set; } = new List<CompositionElement>();
    }
}

File: ClientApp/src/app/api/models/item.ts
/* tslint:disable */
/* eslint-disable */
import { CompositionElement } from '../models/composition-element';
import { ItemTypes } from '../models/item-types';
export interface Item {
  compositions?: null | Array<CompositionElement>;
  id?: number;
  name?: null | string;
  type?: ItemTypes;
}
